crates/core/src/contracts/eth.rs

```diff
@@ -0,0 +1,51 @@
+use std::marker::PhantomData;
+
+use reth_primitives::U256;
+use starknet::core::types::{BlockId, FunctionCall};
+use starknet::providers::Provider;
+use starknet_crypto::FieldElement;
+
+use crate::client::constants::selectors::BALANCE_OF;
+use crate::client::errors::EthApiError;
+use crate::client::helpers::DataDecodingError;
+use crate::models::felt::Felt252Wrapper;
+
+/// Abstraction for the Starknet native token contract.
+pub struct EthContract<'a, P> {
```

Comment:
EthContract is a misleading name, but great change to the PR:D
Also the rustdoc isn't fully accurate, this is rather an abstraction over ERC20.cairo, right? Since we make the contract address agnostic (so could be any erc20)

I suggest:
- CairoErc20
- StarknetErc20
- StarknetFungibleToken